### ISSUE TITLE
Feature Request: Add an easier way to use `new-repl` and `new-exec -- ghci` via a flag?

### ISSUE THREAD

**Issue by ekmett** [ndmitchell/ghcid#214]
It would be really nice if there was some kind of `--new` flag that opted in to using `new-repl` or using a `new-exec` of `ghci`.

As it stands you have to give a long -c to replicate the existing speedup flags, and _which_ command to give seems contingent on whether or not ghcid was given an argument for naming the file or should do a general repl build.

Abstracting over that choice seems to be a plausible thing to ask the tool to do rather than each user.

A significant portion of the complexity of my local build environment comes from trying to figure out how to shim things so that this works somehow, and so that the neovim `:Ghcid` command does the right thing, despite passing the --color argument behind my back.

**Comment by ndmitchell** [comment]
Does `cabal` drop some file down if it's using the `new-` variants? If so, I could just spot that file and use it (I already use the existence of `.stack_work` as an indicator the user has a stack configuration and likes to use it).

**Comment by mitchellwrosen** [comment]
@ndmitchell Well it's not fool-proof of course, but old-cabal has no use for `cabal.project` nor `dist-newstyle/`

**Comment by ndmitchell** [comment]
Running cabal seems that `dist-newstyle` gets created, but not a project file, so I can use that as the trigger.

**Comment by ekmett** [comment]
Sounds good to me. If you're going to add a magic auto-configuration, it'd still be nice to have the option to select between auto-configs. Rather than er.. "stack ranking" them, no pun intended.

**Comment by ndmitchell** [comment]
The auto configurations are fairly nuanced - they do different things based on which files are around, if you pass arguments or not, and I think I'm soon going to have them do different things based on where things like GHC are on your path (e.g. if you have GHC as the selected auto config and its at 8.6.1/bin/ghc, then you might as well turn on colors on the command line rather than waiting until after the first load). My guess is that as long as stack users don't prefer Cabal but test with stack, they will do the right thing. If you want to go away from the auto config I recommend manual config, which takes the magic away entirely.

**Comment by adamse** [comment]
Half the time I cannot even figure out the correct ghcid invocation for new-style cabal. Could a recommended line be added to the readme perhaps?
